packages/dds/sequence/src/test/revertibles.spec.ts

```diff
@@ -169,14 +283,187 @@ describe("Sequence.Revertibles with Remote Edits", () => {
+
+		collection.on("deleteInterval", (interval, local, op) => {
+			appendLocalDeleteToRevertibles(sharedString, interval, revertibles);
+		});
+		collection.removeIntervalById(id);
+		containerRuntimeFactory.processOneMessage();
+
+		collection2.change(id, 3, 8);
+
+		revertIntervalRevertibles(sharedString, revertibles.splice(0));
+		containerRuntimeFactory.processAllMessages();
+
+		assertIntervals(sharedString, collection, [{ start: 0, end: 5 }]);
+		assertIntervals(sharedString2, collection2, [{ start: 0, end: 5 }]);
+	});
+	it("remote interval remove interacting with reverting an interval property change", () => {
+		sharedString.insertText(0, "hello world");
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		const id = collection.add(0, 5, IntervalType.Simple, { foo: "one" }).getIntervalId()!;
+		containerRuntimeFactory.processAllMessages();
+
+		collection.on("propertyChanged", (interval, propertyDeltas, local, op) => {
+			appendLocalPropertyChangedToRevertibles(interval, propertyDeltas, revertibles);
+		});
+		collection.changeProperties(id, { foo: "two" });
+		containerRuntimeFactory.processAllMessages();
+
+		collection2.removeIntervalById(id);
+
+		containerRuntimeFactory.processAllMessages();
+
+		revertIntervalRevertibles(sharedString, revertibles.splice(0));
+		containerRuntimeFactory.processAllMessages();
+
+		assertIntervals(sharedString, collection, []);
+		assertIntervals(sharedString2, collection2, []);
+	});
+	it("remote interval remove interacting with reverting an interval remove", () => {
+		sharedString.insertText(0, "hello world");
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		const id = collection.add(0, 5, IntervalType.Simple).getIntervalId()!;
+		containerRuntimeFactory.processAllMessages();
+
+		collection.on("deleteInterval", (interval, local, op) => {
+			appendLocalDeleteToRevertibles(sharedString, interval, revertibles);
+		});
+		collection.removeIntervalById(id);
+		containerRuntimeFactory.processAllMessages();
+
+		collection2.removeIntervalById(id);
+
+		revertIntervalRevertibles(sharedString, revertibles.splice(0));
+		containerRuntimeFactory.processAllMessages();
+
+		assertIntervals(sharedString, collection, [{ start: 0, end: 5 }]);
+		assertIntervals(sharedString2, collection2, [{ start: 0, end: 5 }]);
+	});
+	it("remote interval property change interacting with reverting an interval remove", () => {
+		sharedString.insertText(0, "hello world");
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		const id = collection.add(0, 5, IntervalType.Simple, { foo: "one" }).getIntervalId()!;
+		containerRuntimeFactory.processAllMessages();
+
+		collection.on("deleteInterval", (interval, local, op) => {
+			appendLocalDeleteToRevertibles(sharedString, interval, revertibles);
+		});
+		collection.removeIntervalById(id);
+		containerRuntimeFactory.processAllMessages();
+
+		collection2.changeProperties(id, { foo: "two" });
```

Comment:
Same thing here: client 2 has already seen the remove. Just get rid of the processAllMessages right before this. It might maybe be more interesting to have the server sequence the changeProperties before the remove. Either way, the remote property change gets thrown away, I think, which is fine.

Reply:
I meant to remove the processAllMessages above, not below. The way it is now, `collection2.changeProperties(id, { foo: "two" });` doesn't do anything because client 2 has already applied the interval remove from client 1.